splinepy/bspline.py

```diff
@@ -92,11 +126,27 @@ def insert_knots(self, parametric_dimension, knots):
                 "(Too small)"
             )
 
+        if return_matrix:
+            data = splinepy_core.global_knot_insertion_matrix(
+                self.knot_vectors,
+                self.degrees,
+                parametric_dimension,
+                utils.data.enforce_contiguous(knots, dtype="float64"),
+                settings.TOLERANCE,
+            )
+            if has_scipy:
+                matrix = scipy.sparse.csr_matrix(data[0], shape=data[1])
+            else:
+                matrix = np.zeros(data[1])
+                matrix[data[0][1][0], data[0][1][1]] = data[0][0]
+
```

Comment:
I thought this might be more intuitive, if we call insert_knots, then knots are inserted. We could also outsource this to another function, e.g. knot_insertion_matrix to avoid confusion.

Reply:
Has been moved